database/service_db_test.go

```diff
@@ -30,13 +30,28 @@ func TestServiceDBSave(t *testing.T) {
 	db, closer := openServiceDB(t)
 	defer closer()
 
-	s := &service.Service{ID: "00", Alias: "1", Name: "test-service"}
-	require.NoError(t, db.Save(s))
-	s = &service.Service{ID: "00", Alias: "2", Name: "test-service"}
-	require.NoError(t, db.Save(s))
+	s1 := &service.Service{ID: "00", Alias: "1", Name: "test-service"}
+	require.NoError(t, db.Save(s1))
+
+	// save same service. should replace
+	require.NoError(t, db.Save(s1))
+	ss, _ := db.All()
+	require.Len(t, ss, 1)
+
+	// different id, same alias. should replace s1
+	s2 := &service.Service{ID: "01", Alias: "1", Name: "test-service"}
+	require.NoError(t, db.Save(s2))
+	_, err := db.Get(s1.ID)
+	require.IsType(t, &ErrNotFound{}, err)
+
+	// different id, different alias. should not replace anything
+	s3 := &service.Service{ID: "02", Alias: "2", Name: "test-service"}
+	require.NoError(t, db.Save(s3))
+	ss, _ = db.All()
+	require.Len(t, ss, 2)
```

Comment:
it'd be better also to compare received services here with Equal so we can make sure that we really overwrite the service with new one(s2) correctly.